sky/backends/backend_utils.py

```diff
@@ -481,226 +481,85 @@ def add_cluster(
-            config = f.readlines()
-
         proxy_command = auth_config.get('ssh_proxy_command', None)
+        docker_proxy_command = None
+        head_port = ports[0]
         if docker_user is not None:
             docker_proxy_command_generator = lambda ip: ' '.join(
                 ['ssh'] + command_runner.ssh_options_list(key_path, None) +
                 ['-W', '%h:%p', f'{auth_config["ssh_user"]}@{ip}'])
-        docker_proxy_command = None
+            head_port = constants.DEFAULT_DOCKER_PORT
 
-        # Check if ~/.ssh/config contains existing names
-        host_lines = [f'Host {c_name}' for c_name in worker_names]
-        for i, line in enumerate(config):
-            if line.strip() in host_lines:
-                idx = host_lines.index(line.strip())
-                prev_line = config[i - 1] if i > 0 else ''
-                logger.warning(f'{cls.ssh_conf_path} contains '
-                               f'host named {worker_names[idx]}.')
-                host_name = external_worker_ips[idx]
-                logger.warning(f'Using {host_name} to identify host instead.')
-                ip = external_worker_ips[idx]
-                if docker_user is not None:
-                    docker_proxy_command = docker_proxy_command_generator(ip)
-                    ip = 'localhost'
-                # TODO(romilb): Update port number when k8s supports multinode
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, host_name, ip, username, key_path,
-                    proxy_command, port, docker_proxy_command)
-
-        # All workers go to SKY_USER_FILE_PATH/ssh/{cluster_name}
-        for i, line in enumerate(extra_config):
-            if line.strip() in host_lines:
-                idx = host_lines.index(line.strip())
-                prev_line = extra_config[i - 1] if i > 0 else ''
-                if prev_line.strip().startswith(sky_autogen_comment):
-                    host_name = worker_names[idx]
-                    overwrites[idx] = True
-                    overwrite_begin_idxs[idx] = i - 1
-                ip = external_worker_ips[idx]
-                if docker_user is not None:
-                    docker_proxy_command = docker_proxy_command_generator(ip)
-                    ip = 'localhost'
-                # TODO(romilb): Update port number when k8s supports multinode
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, host_name, ip, username, key_path,
-                    proxy_command, port, docker_proxy_command)
-
-        # This checks if all codegens have been created.
-        for idx, ip in enumerate(external_worker_ips):
+        # Start generating the config with the head node
+        codegen = cls._get_generated_config(
+            sky_autogen_comment, host_name, ip, username, key_path,
+            proxy_command, head_port, docker_proxy_command_generator(ip)) + '\n'
```

Comment:
This `docker_proxy_command_generator` is undefined here. Please refer to the refactoring below.

Reply:
I have updated the code. Still haven't run the checks that you mentioned. I was curious about what is the syntax for re-launching a given cluster.